### PR TITLE
Feature - Implement OAuth 2.0 Authorization Code Grant login flow

### LINKED ISSUE #1: Implement OAuth Authorization Code Grant flow logins
Branch created: `feature/implement-oauth-code-grant-flow-logins-in-auth-package_CU-d0tcyq` from `master`

### PR DESCRIPTION
**Pull request by adarleyjrr**
Closes #1 #d0tcyq

### CODE REVIEW

File: src/service.ts [hunk 1]
```diff
@@ -399,8 +399,8 @@ export const authService = (
 
               // If we get here we succeeded
               sessionPayload = getSessionFromAuth({
-                ...userAuth,
-                refreshToken: sessionPayload.refreshToken
+                refreshToken: sessionPayload.refreshToken,
```
Comment (AssisrMatheus) [inline src/service.ts:402]: Is this change really necessary? In reality, the refresh token should never change, if that is the case something has happened(like a password change or the user asked to be logged out of all devices) and the user needs to be logged out instead. So we should always keep the same refreshToken. And that's why it was overwriting the userAuth, if it ever returns a new refresh token.

**Review by AssisrMatheus — CHANGES_REQUESTED** [review]
Looks good! You might just want to raise the version in package.json. So that's why i'm requesting a change. But everything else is great.

File: src/service.ts (hunk 1 above)
Comment (adarleyjrr) [reply src/service.ts:402]: I ran into the issue that led me to changing this when the OAuth server library that we have in the auth project would make a new access token, by default it was revoking the old refresh token and generating a new one, alongside with the access token. Then I made a small research and found out that a lot of companies that implement OAuth login flows revoke the refresh token and generate a new one every time you use that refresh token, and since that's also the default behaviour of the server library that I used I assumed this to be a great security feature.

The way I edited this behaviour allows the developer that uses the library to choose if he wants to refresh the refresh token or not in the `onRefresh` callback.

But I can override the default behaviour of the oauth server library and just create a new access token, keeping the old refresh token if you don't think this is a great idea. I know a downside of refreshing the refresh token is when a network error happens, the user will basically be logged out because the refresh token will be revoked and the client won't be able to get the new one.

Comment (AssisrMatheus) [reply src/service.ts:402]: I see! That makes sense, it's fine, just wanted to know also so I can learn more. Good work
